optuna/integration/allennlp.py

```diff
@@ -0,0 +1,86 @@
+import json
+import os
+from typing import Any
+from typing import Dict
+from typing import List
+
+import optuna
+
+try:
+    import _jsonnet
+    import allennlp.commands
+    import allennlp.common.util
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    _available = False
+    TrackerCallback = object
+
+
+class AllenNLPExecutor(object):
+    """Allennlp extension to use optuna with an allennlp config file.
+
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` corresponding to the current evaluation
+            of the objective function.
+        config_file:
+            An allennlp config file.
+            Hyperparameters should be masked with `std.extVar`.
+            Please refer to `the config example <https://github.com/allenai/allentune/blob/
+            master/examples/classifier.jsonnet>`_.
+        serialization_dir:
+            A path which model weights and logs are saved.
+        metrics:
+            An evaluation metric for the result of ``objective``.
+
+    """
+
+    def __init__(
+        self,
+        trial: optuna.Trial,
+        config_file: str,
+        serialization_dir: str,
+        metrics: str = "best_validation_accuracy",
+        *,
+        include_package: List[str] = []
+    ):
+
+        self._params = trial.params
+        self._config_file = config_file
+        self._serialization_dir = serialization_dir
+        self._metrics = metrics
+        self._include_package = include_package
+
+    def _build_params(self) -> Dict[str, Any]:
+        """Create a dict of params for allennlp."""
+
+        for key, value in self._params.items():
+            self._params[key] = str(value)
+        _params = json.loads(_jsonnet.evaluate_file(self._config_file, ext_vars=self._params))
+        return allennlp.common.params.infer_and_cast(_params)
+
+    def run(self) -> float:
+        """Train a model using allennlp."""
+
+        for package_name in self._include_package:
+            allennlp.common.util.import_module_and_submodules(package_name)
```

Comment:
I think any test cases do not visit this line. Please add test cases.

Reply:
I added tests in 587db54 and found there is a breaking change in this feature.

`import_module_and_submodule` will be introduced in the next release allennlp,
which is the renamed method of `import_modules` in `v0.9.0`.